src/haz3lweb/Update.re

```diff
@@ -312,12 +312,15 @@ let apply =
       | DebugLoad => failwith("impossible")
       | Scratch(_) => Error(FailedToSwitch) // one editor per scratch
       | School(m, specs, exercise) =>
-        let exercise = SchoolExercise.switch_editor(n, exercise);
+        let exercise = ref(exercise);
```

Comment:
why are you creating a ref here? not sure why this was changed...

Reply:
So it's to solve the issue that it crashes when jump to an non-existing editor under the student mode.
The logic is:
if it's student mode and we are jumping to the hidden editors, don't modify exercise.
else change exercise to SchoolExercise.switch_editor(n, exercise);.
So the ref here is made to change the exercise inside the if statement and be used later.

Reply:
You don't need a ref here, you can just bind to a variable...

```
let exercise = 
  if (model.settings...) { 
    SchoolExercise.switch...
  } else { exercise } 
```